pkg/controllers/song.go

```
@@ -52,14 +60,24 @@ func (ctrl *Controller) AddSong(ctx context.Context, req *utils.AppReq) utils.Ap
 
 // GetAllSong(context.Context, *utils.AppReq) function calls a helper GetAllSong function to get all songs from database and returns a utils.AppResp response containing error message, status code and data
 func (ctrl *Controller) GetAllSong(ctx context.Context, req *utils.AppReq) utils.AppResp {
-	user := ctx.Value("user").(string)
+	tokenClaims := ctx.Value("token").(map[string]interface{})
+	user := tokenClaims["user"].(string)
 	var song []models.Song
+	if val, err := ctrl.SongCache.Get(user); err == nil {
+		json.Unmarshal([]byte(val), &song)
+		return utils.AppResp{
+			"data":   song,
+			"status": http.StatusOK,
+		}
+	}
 	if err := ctrl.SongRepo.GetAllSong(&song, user); err != nil {
 		return utils.AppResp{
 			"error":  err.Error(),
 			"status": http.StatusInternalServerError,
 		}
 	}
+	val, _ := json.Marshal(song)
```

any specific reason for ignoring the error.

---

Not such reason. Will update in the upcoming commit.